django/db/models/fields/related_lookups.py

```diff
@@ -83,6 +83,29 @@ def as_sql(self, compiler, connection):
         else:
             return super(RelatedIn, self).as_sql(compiler, connection)
 
+    def __getstate__(self):
+        from django.db.models.query import QuerySet
+        obj_dict = self.__dict__.copy()
+        del obj_dict['rhs']
+        if isinstance(self.rhs, QuerySet):
+            # Pickle query fron queryset only
+            return (self.rhs.model, self.rhs.query, obj_dict)
+        else:
+            # Pickle RHS as is
+            return (None, self.rhs, obj_dict)
+
+    def __setstate__(self, state):
+        model, rhs, obj_dict = state
+        if model is None:
+            # Unpickle full RHS
+            self.rhs = rhs
+        else:
+            # Unpickle RHS as query
+            queryset = model.objects.none()
+            queryset.query = rhs
+            self.rhs = queryset
```

Comment:
I guess that would work just fine.

But about the query holder -- you're right, that code fails if there is no manager called "objects", or model has objects manager but it's not the one used with query.

Hopefully  query knows where it actually came from so I can reconstruct it properly.